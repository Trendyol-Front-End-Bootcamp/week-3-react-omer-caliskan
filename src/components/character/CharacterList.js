import CharacterCard from './CharacterCard';
import { Link } from 'react-router-dom'
import '../../assets/css/characterList.css';
import React from 'react';

function CharacterList({characters}) {

    return (  
        <div className="list">
            {characters.map((char, id) => (
                <Link key={char.id} to={`/${char.id}`}>
                    <CharacterCard key={id} char={char} />
                </Link>
            ))}
        </div>
    );
}

export default CharacterList;