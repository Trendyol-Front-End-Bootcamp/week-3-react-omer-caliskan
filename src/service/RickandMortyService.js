const axios = require('axios')

export const getCharacterById = async (charId) => {

  let character
  
  await axios
      .get(`https://rickandmortyapi.com/api/character/${charId}`)
      .then(({data}) => {character = data})

      return character;
}

export const getCharactersByPageandFilter = async (page, gender, status) => {

  let characters = []
  
  let query = `?page=${page}${gender!=="All" ? "&gender=" + gender : ""}${status!=="All" ? "&status=" + status : ""}`;

  await axios
      .get(`https://rickandmortyapi.com/api/character/${query}`)
      .then(({data}) => {characters = data})

  return characters;
}

export const getEpisodeName = async (episodesUrl) => {

  let episodeName;

  await axios
        .get(`${episodesUrl}`)
        .then(({data}) => {episodeName = data.name})
  
  return episodeName;
}