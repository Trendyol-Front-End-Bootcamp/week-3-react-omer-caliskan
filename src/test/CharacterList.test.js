import { getCharactersByPageandFilter } from "../service/RickandMortyService";
import React from 'react'
import Enzyme, { shallow } from 'enzyme';
import Adapter from "enzyme-adapter-react-16";
import CharacterList from "../components/character/CharacterList";


Enzyme.configure({ adapter: new Adapter() })

describe("Character List Component", () => {
        let characterList
        beforeEach(async () => {
          const characters = await getCharactersByPageandFilter(1, "All", "All")

          characterList = shallow(<CharacterList characters={[characters]} />)
        })
        it("should render character list", () => {
          expect(characterList).toMatchSnapshot()
        })
  
        it("should check character list class", () => {
          const checkClass = characterList.hasClass("list")
  
          expect(checkClass).toBeTruthy()
        })
  
        it("should check character card called from character list", () => {
          const checkCalled = characterList.find("Character Card")
  
          expect(checkCalled).toBeTruthy()
        })
  
        it("should check character cards have link", () => {
          const checkLink = characterList.find("Link")
  
          expect(checkLink).toBeTruthy()
        })
  
        it("should check all character cards have right link by its id", () => {
          const checkLinkByItem = characterList.find("Link").props().children.props.char.id
          const checkLinkByLink = characterList.find("Link").prop('to')
  
          expect("/" + checkLinkByItem).toEqual(checkLinkByLink)
        })
      })