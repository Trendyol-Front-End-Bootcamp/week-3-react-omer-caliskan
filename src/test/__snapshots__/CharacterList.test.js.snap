// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Character List Component should render character list 1`] = `
"<div className=\\"list\\">
  <Link to=\\"/undefined\\">
    <CharacterCard char={{...}} />
  </Link>
</div>"
`;
